enter.c

```diff
@@ -225,13 +226,65 @@ int enter(struct entry_settings *opts)
 		}
 	}
 
+	int rootpid = getpid();
+	bool req_cgroup_clean = false;
+	int cleanfd;
+
+	// Do not join the root cgroup if limits are to be applied, otherwise bst needs to create sub-hierarchies
+	if (!opts->nactiveclimits && opts->cgroup.present) {
```

Comment:
It's better to be explicit in intent for quantities -- I'd use `if (opts->nactiveclimits > 0 && opts->cgroup.present)`. This ensures that this stays correct even if the type of the quantity changes to some signed value for whatever reason. Explicit is better than clever, and there has been many situations where bugs have been introduced because of the implicit meaning of nonzero to mean true.